### DISCUSSION TITLE
Is the default value for versioning in RegexManagers listed in the documentation incorrect?

### DISCUSSION THREAD

**Discussion by mokuno441** [renovatebot/renovate#22957] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.89.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I believe the default value for versioning in the documented [RegexManagers](https://docs.renovatebot.com/modules/manager/regex/#required-fields) is `semver`, not `semver-coerced`.

### What you tried so far.

In [Renovate Docs > Configuration Options > regexManagers](https://docs.renovatebot.com/configuration-options/#regexmanagers), versioning defaults to `semver`.  
>We recommend that you also tell Renovate what versioning to use. If the versioning field is missing, then Renovate defaults to using semver versioning.

However, in [Renovate Docs > Custom Manager Support using Regex > Required Fields](https://docs.renovatebot.com/modules/manager/regex/#required-fields), it is listed as `semver-coerced`.
>You can optionally have a versioning capture group or a versioningTemplate config field. If neither are present, semver-coerced will be used as the default

So, to verify the default value, I ran it on [this repository](https://github.com/mokuno441/renovate-test-repo) and it appeared that `semver` was used.

Therefore, I believe the default value for versioning in the documented [RegexManagers](https://docs.renovatebot.com/modules/manager/regex/#required-fields) is `semver`, not `semver-coerced`.

--
Finally, I am not sure if I am communicating well because my English is not good, but I would appreciate your help.

**Comment by secustor** [comment]
Yes that is correct, we changed the default value from `semver` to `semver-coerced` in [v35.0.0](https://github.com/renovatebot/renovate/releases/tag/35.0.0) and this seems to be a left over.

PRs welcome.

**Reply by rarkins** [reply]
@mokuno441 How did you verify that semver is used and not coerced?

**Reply by mokuno441** [reply]
@rarkins 
I have created a [release](https://github.com/mokuno441/renovate-test-repo/releases/tag/v1.1) that is not strictly a semantic version, and running renovate with the default values does not create a PR, which confirms that semver is being used.

**Reply by mokuno441** [reply]
I then set `versioningTemplate` to `semver-coerced` and a PR was created for this release.

So I believe the actual default value is `semver`.

**Reply by rarkins** [reply]
OK, please create a reproduction repo if you'd like us to look at it

**Reply by mokuno441** [reply]
@rarkins 
I have created a reproducible repository.
https://github.com/mokuno441/mokuno441-renovate

**Reply by HonkingGoose** [reply]
That reproduction is for `renovate/renovate:35.89.1`, but the latest version is `35.144.0`. Maybe you should also test the latest version in a new reproduction repository?

**Reply by mokuno441** [reply]
@HonkingGoose Thanks for pointing that out. I think you are right, we need to reproduce the issue with the latest version.

I changed the docker image tag in README.md to `latest`.
Then I ran it again and confirmed that the version of renovate is 35.144.0 and that the results are the same.

**Reply by rarkins** [reply]
An extract dry run on the reproduction repo produces this:

```
 INFO: Extracted dependencies (repository=mokuno441/mokuno441-renovate)
       "packageFiles": {
         "regex": [
           {
             "deps": [
               {
                 "depName": "mokuno441/renovate-test-repo",
                 "currentValue": "v1.1",
                 "datasource": "github-releases",
                 "versioning": "semver-coerced",
                 "replaceString": "renovate-test-repo_version=v1.1"
               }
             ],
             "matchStrings": ["renovate-test-repo_version=(?<currentValue>.*)"],
             "depNameTemplate": "mokuno441/renovate-test-repo",
             "datasourceTemplate": "github-releases",
             "versioningTemplate": "semver-coerced",
             "packageFile": "versions.ini"
           },
           {
             "deps": [
               {
                 "depName": "mokuno441/renovate-test-repo-2",
                 "currentValue": "v1.1",
                 "datasource": "github-releases",
                 "replaceString": "renovate-test-repo-2_version=v1.1"
               }
             ],
             "matchStrings": ["renovate-test-repo-2_version=(?<currentValue>.*)"],
             "depNameTemplate": "mokuno441/renovate-test-repo-2",
             "datasourceTemplate": "github-releases",
             "packageFile": "versions.ini"
           }
         ]
       }
```

I think what we're seeing is that the default _datasource_ versioning of `semver` gets applied: https://github.com/renovatebot/renovate/blob/a7f6514ebe2f9de2fda4aacaf5ac8b20173768ae/lib/modules/datasource/index.ts#L239-L251

There's not many hits for the `semver-coerced` string in the code either:

![image](https://github.com/renovatebot/renovate/assets/6311784/710b5374-205b-4c51-a1a5-070a23426143)

My conclusion: `semver` seems to be the default, not `semver-coerced`. I wonder if the docs for the other package managers is wrong too.

**Reply by HonkingGoose** [reply]
I'll let you maintainers figure out what's right and wrong, and let you update the docs and code. 😉

**Reply by secustor** [reply]
https://github.com/renovatebot/renovate/pull/23020

**Reply by rarkins** [reply, marked as answer]
Answer: value is incorrect, will be corrected in v36
